save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/cli/DiktatCliRunner.kt

```diff
@@ -32,57 +32,41 @@ import kotlin.io.path.*
 @Component
 class DiktatCliRunner(
     private val toolStorage: ToolStorage,
-) : CliRunner<DiktatDemoAdditionalParams, DiktatDemoResult> {
-    override fun getExecutable(workingDir: Path, params: DiktatDemoAdditionalParams): Path {
-        val toolName = params.tool.name.lowercase()
-        val version = if (params.tool == DiktatDemoTool.DIKTAT) {
-            DIKTAT_VERSION
-        } else {
-            KTLINT_VERSION
+) : CliRunner<DiktatDemoAdditionalParams, ToolKey, DiktatDemoResult> {
+    private fun getRunCommandForDiktat(
+        workingDir: Path,
+        testPath: Path,
+        outputPath: Path,
+        configPath: Path?,
+        params: DiktatDemoAdditionalParams,
+    ): String = buildString {
+        // TODO: this information should not be hardcoded but stored in database
+        val ktlintExecutable = getExecutable(workingDir, DiktatDemoTool.KTLINT.toToolKey("ktlint"))
+        val diktatExecutable = getExecutable(workingDir, DiktatDemoTool.DIKTAT.toToolKey("diktat-1.2.3.jar"))
```

Comment:
And here, `1.2.3` is the **version** (as opposed to the VCS tag), so please extract it and store alongside the **tag**.

Reply:
The point is that executable name is fetched from GitHub so we do not need to hardcode it, either database should be used to store this kind of information, or storage itself (e.g. list or something like that). Database is needed in order to keep all the data to construct the URL for download as well as dependencies (like it is in `diktat` - it requires a special version of `ktlint`)

Reply:
Fair enough.